Resources/public/js/views/fields/ez-isbn-editview.js

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-isbn-editview', function (Y) {
+    "use strict";
+    /**
+     * Provides the field edit view for the ISBN field
+     *
+     * @module ez-isbn-editview
+     */
+    Y.namespace('eZ');
+
+    var FIELDTYPE_IDENTIFIER = 'ezisbn';
+
+    /**
+     * ISBN edit view
+     *
+     * @namespace eZ
+     * @class ISBNEditView
+     * @constructor
+     * @extends eZ.FieldEditView
+     */
+    Y.eZ.ISBNEditView = Y.Base.create('isbnEditView', Y.eZ.FieldEditView, [], {
+
+        events: {
+            '.ez-isbn-input-ui input': {
+                'blur': 'validate',
+                'valuechange': 'validate',
+            },
+        },
+
+        /**
+         * Validates the current input of the ISBN field
+         *
+         * @method validate
+         */
+        validate: function () {
+            this.set('errorStatus', false);
+            if ( this.get('fieldDefinition').isRequired && this._isFieldEmpty() ){
+                this.set('errorStatus', 'This field is required');
+            } else {
+                if (this.get('fieldDefinition').fieldSettings.isISBN13) {
+                    this._checkISBN13();
+                } else {
+                    this._checkISBN10();
+                }
+            }
+        },
+
+        /**
+         * Checks whether the field is empty
+         *
+         * @method _isFieldEmpty
+         * @protected
+         * @return {Boolean}
+         */
+        _isFieldEmpty: function () {
+            var inputString = this.get('container').one('.ez-isbn-input-ui input').get('value'),
+                rawInputString = inputString.replace(/-/g, "");
```

Comment:
this is done several times, you should introduce a method to do that (`_cleanDashes` or something like that)